src/libraries/System.Private.CoreLib/src/System/String.Manipulation.cs

```diff
@@ -1446,7 +1446,7 @@ private string[] SplitOmitEmptyEntries(ReadOnlySpan<int> sepList, ReadOnlySpan<i
 
             for (int i = 0; i < numReplaces && currIndex < Length; i++)
             {
-                if (sepList[i] - currIndex > 0)
+                if (sepList[i] > currIndex)
```

Comment:
This could have been written such that `sepList[i] - currIndex` gets CSEd with `sepList[i] - currIndex` on the next line. Could you please verify that the JITed code looks better with this change?